lib/jxl/dec_reconstruct.cc

```diff
@@ -37,141 +37,114 @@ HWY_BEFORE_NAMESPACE();
 namespace jxl {
 namespace HWY_NAMESPACE {
 
-Status UndoXYBInPlace(Image3F* idct, const Rect& rect,
+template <typename Op>
```

Comment:
I think there's some potential here to reduce the binary size by removing this template and using a virtual function. It won't be inlined of course but given that we only do a Store afterwards I don't think it will make much difference in runtime performance. I'll do this in a follow up.